<a href="https://colab.research.google.com/github/Mazridha3103/Mazridha3103/blob/main/Mazridha_Achiri_RAL.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Rancangan Acak Lengkap (RAL) 

Rancangan Acak Lengkap merupakan jenis rancangan percobaan dimana perlakuan diberikan secara acak kepada seluruh unit percobaan. Hal ini dapat dilakukan karena lingkungan tempat percobaan diadakan relatif homogen sehingga media atau tempat percobaan tidak memberikan pengaruh berarti pada respon yang diamati.  Untuk melihat apakah hasil respon dari percobaan dengan rancangan acak lengkap ini berbeda atau tidak antar perlakuan, maka kita bisa menggunakan analisis One Way ANOVA. Hipotesis statistik yang diuji pada one way Anova adalah sebagai berikut:

H<sub>0</sub>: π<sub>1</sub>=π<sub>2</sub>=π<sub>3</sub>=.....=π<sub>k</sub>

H<sub>1</sub>: Minimal terdapat satu rata-rata perlakuan tertentu yang berbeda dengan perlakuan yang lain.

Dimana k adalah banyaknya perlakuan. jika pada hasil ANOVA menunjukkan nilai p-value < tingkat kesalahan α (biasanya digunakan 5%) yang kita inginkan, maka kita menolak H<sub>0</sub> sehingga menerima H<sub>1</sub>. Sehingga jika p-value ≥ tingkat kesalahan α (biasanya digunakan 5%) yang kita inginkan, maka kita menerima H<sub>0</sub>.


# Input Data


In [ ]:
# Data analisa pH dibuat dalam bentuk list permasing-masing perlakuan
perlakuan1=[2.81,2.81,2.95,2.95] 
perlakuan2=[2.78,2.78,2.74,2.74]
perlakuan3=[2.72,2.74,2.74,2.75]
perlakuan4=[2.83,2.80,2.76,2.76]
print(perlakuan1,perlakuan2,perlakuan3,perlakuan4)

[2.81, 2.81, 2.95, 2.95] [2.78, 2.78, 2.74, 2.74] [2.72, 2.74, 2.74, 2.75] [2.83, 2.8, 2.76, 2.76]


Data Uji analisa pH didapatkan dari Uji analisa pH ekstrak Parijoto dengan 4 perlakuan yaitu dengan volume cairan yang berbeda- beda

# Analisis One Way ANOVA

In [ ]:
# import library scipy.stats dengan nama ss
import scipy.stats as ss
#panggil fungsi One Way ANOVA
ss.f_oneway(perlakuan1,perlakuan2,perlakuan3,perlakuan4)

F_onewayResult(statistic=7.479125248508941, pvalue=0.0043946364497371125)

Berdasarkan hasil tersebut dapat dilihat bahwa p-value < 0.05, sehingga tolak H<sub>0</sub>. Dapat di interpretasikan bahwa dari 4 perlakuan yang digunakan dalam percobaan, minimal terdapat 1 perlakuan yang hasilnya berbeda dengan perlakuan lainnya.

# Uji Lanjut Tukey

Jika pada hasil analisis pada One Way ANOVA adalah tolak H<sub>0</sub>, maka dapat disimpulkan bahwa ada minimal salah satu perlakuan yang memberikan hasil rata-rata yang berbeda dengan perlakuan lain. Perbedaan detail terkait hasil rata-rata perlakuan tidak dapat ditentukan secara langsung. Oleh karena itu, diperlukan uji lanjut untuk mengetahui hasil dari rata-rata tersebut berbeda ataukah sama. Salah satu uji lanjut yang sering digunakan adalah Uji Tukey. Hipotesis yang di Uji adalah sebagai berikut :

H<sub>0</sub>: π<sub>i</sub>=π<sub>j</sub>

H<sub>1</sub>: π<sub>i</sub> \ne π<sub>j</sub>


Jika pada hasil Uji Tukey menunjukkan nilai p-value < tingkat kesalahan α (biasanya digunakan 5%) yang kita inginkan, maka kita menolak H<sub>0</sub> sehingga menerima H<sub>1</sub>. Sehingga jika p-value ≥ tingkat kesalahan α (biasanya digunakan 5%) yang kita inginkan, maka kita menerima H<sub>0</sub>.

In [ ]:
#import library yang dibutuhkan
import pandas as pd
import numpy as np
import statsmodels.stats.multicomp as ssn
df=pd.DataFrame({'score':perlakuan1+perlakuan2+perlakuan3+perlakuan4,'group':np.repeat(['perlakuan1','perlakuan2','perlakuan3','perlakuan4'], repeats=4)})
print(df)

    score       group
0    2.81  perlakuan1
1    2.81  perlakuan1
2    2.95  perlakuan1
3    2.95  perlakuan1
4    2.78  perlakuan2
5    2.78  perlakuan2
6    2.74  perlakuan2
7    2.74  perlakuan2
8    2.72  perlakuan3
9    2.74  perlakuan3
10   2.74  perlakuan3
11   2.75  perlakuan3
12   2.83  perlakuan4
13   2.80  perlakuan4
14   2.76  perlakuan4
15   2.76  perlakuan4


/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [ ]:
hasiltukey=ssn.pairwise_tukeyhsd(endog=df['score'],groups=df["group"],alpha=0.05)
print(hasiltukey)

    Multiple Comparison of Means - Tukey HSD, FWER=0.05     
  group1     group2   meandiff p-adj   lower   upper  reject
------------------------------------------------------------
perlakuan1 perlakuan2    -0.12 0.0138 -0.2161 -0.0239   True
perlakuan1 perlakuan3  -0.1425 0.0041 -0.2386 -0.0464   True
perlakuan1 perlakuan4  -0.0925 0.0606 -0.1886  0.0036  False
perlakuan2 perlakuan3  -0.0225 0.8972 -0.1186  0.0736  False
perlakuan2 perlakuan4   0.0275 0.8149 -0.0686  0.1236  False
perlakuan3 perlakuan4     0.05 0.4446 -0.0461  0.1461  False
------------------------------------------------------------


Berdasarkan hasil tersebut bahwa kita dapat melihat bahwa:
1. Pada perlakuan 1 vs perlakuan 2, dapat dilihat p-value (0.0138) < α (0.05), sehingga terima H<sub>0</sub> yang berarti rata-rata perlakuan 1 berbeda dengan rata-rata perlakuan 2. Perbedaan rata-rata perlakuan 1 dengan rata-rata 0.12 atau negatif mengindikasikan bahwa perlakuan 1 memberikan rata-rata score lebih tinggi daripada perlakuan 2.
2. Pada perlakuan 1 vs perlakuan 3, dapat dilihat p-value (0.0041) < α (0.05), sehingga terima H<sub>0</sub> yang berarti rata-rata perlakuan 1 berbeda dengan rata-rata perlakuan 3. Perbedaan rata-rata perlakuan 1 dengan rata-rata 0.1425 atau negatif mengindikasikan bahwa perlakuan 1 memberikan rata-rata score lebih tinggi daripada perlakuan 3.
3. Pada perlakuan 1 vs perlakuan 4, dapat dilihat p-value (0.0606) > α (0.05), sehingga terima H<sub>0</sub> yang berarti rata-rata perlakuan 1 sama dengan rata-rata perlakuan 4.
4. Pada perlakuan 2 vs perlakuan 3, dapat dilihat p-value (0.8972) > α (0.05), sehingga terima H<sub>0</sub> yang berarti rata-rata perlakuan 2 sama dengan rata-rata perlakuan 3.
5. Pada perlakuan 2 vs perlakuan 4, dapat dilihat p-value (0.8149) > α (0.05), sehingga terima H<sub>0</sub> yang berarti rata-rata perlakuan 2 sama dengan rata-rata perlakuan 4.
6. Pada perlakuan 3 vs perlakuan 4, dapat dilihat p-value (0.04446) < α (0.05), sehingga tolak H<sub>0</sub> yang berarti rata-rata perlakuan 3 berbeda dengan rata-rata perlakuan 4. Perbedaan rata-rata perlakuan 1 dengan rata-rata 0.05 atau positif mengindikasikan bahwa perlakuan 4 memberikan rata-rata score lebih tinggi daripada perlakuan 3



Berdasarkan hasil tersebut kita dapat menyimpulkan bahwa perlakuan 2 dan perlakuan 3, dapat dianggap memberikan hasil skor yang relatif sama (Keduanya kita masukkan dalam kelompok A). Perlakuan 1 dengan perlakuan 4 juga dianggap memberikan hasil skor yang relatif sama (Keduanya kita masukkan kedalam kelompok B). Perlakuan 1 dan perlakuan 4 memberikan hasil skor yang relatif lebih tinggi dari pada perlakuan 2 dan perlakuan 4. jika tujuan kita ingin mendapatkan hasil skor yang paling baik maka kita bisa menggunakan perlakuan 1 ataupun perlakuan 4 
